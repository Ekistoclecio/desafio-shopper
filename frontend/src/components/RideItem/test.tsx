import { fireEvent, screen, waitFor } from '@testing-library/react';

import { RideItem } from '.';

import { renderWithTheme } from '@/utils/tests/helpers';

describe('<RideItem />', () => {
  it('should render the RideItem', () => {
    const { container } = renderWithTheme(
      <RideItem
        driver={{
          id: 1,
          name: 'Teste Motorista',
          description: 'Teste Descrição',
          value: 10,
          vehicle: 'Teste Veículo',
          review: {
            rating: 4,
            comment: 'Teste Comentário',
          },
        }}
        onClickSolicitation={() => {}}
      />,
    );

    expect(screen.getByText('Teste Motorista')).toBeInTheDocument();
    expect(screen.getByText('Teste Descrição')).toBeInTheDocument();
    expect(screen.getByText('Teste Veículo')).toBeInTheDocument();
    expect(screen.getByText('R$ 10')).toBeInTheDocument();

    waitFor(() => {
      expect(screen.getByRole('region')).toBeInTheDocument();
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  //tooltip test
  it('should render the tooltip', () => {
    renderWithTheme(
      <RideItem
        driver={{
          id: 1,
          name: 'Teste Motorista',
          description: 'Teste Descrição',
          value: 10,
          vehicle: 'Teste Veículo',
          review: {
            rating: 4,
            comment: 'Teste Comentário',
          },
        }}
        onClickSolicitation={() => {}}
      />,
    );

    const ratingElement = screen.getByText('Teste Motorista');

    fireEvent.mouseOver(ratingElement);

    waitFor(() => {
      expect(screen.getByText('Teste Comentário')).toBeInTheDocument();
    });
  });

  it('should call onClickSolicitation when clicked', () => {
    const onClick = jest.fn();

    renderWithTheme(
      <RideItem
        driver={{
          id: 1,
          name: 'Teste Motorista',
          description: 'Teste Descrição',
          value: 10,
          vehicle: 'Teste Veículo',
          review: {
            rating: 4,
            comment: 'Teste Comentário',
          },
        }}
        onClickSolicitation={onClick}
      />,
    );

    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
