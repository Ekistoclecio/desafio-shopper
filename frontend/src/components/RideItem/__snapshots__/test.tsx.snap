// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideItem /> should render the RideItem 1`] = `
<div
  class="MuiBox-root css-1huvmu8"
>
  <div
    class="MuiBox-root css-1omog99"
  >
    <div
      class="MuiBox-root css-1km0bek"
    >
      <div
        class="MuiBox-root css-1918gzb"
        data-mui-internal-clone-element="true"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-l4er83-MuiSvgIcon-root"
          data-testid="InfoIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-114o3mv-MuiTypography-root"
        >
          Teste Motorista
        </p>
      </div>
      <div
        class="MuiBox-root css-1918gzb"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DirectionsCarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16m11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5M5 11l1.5-4.5h11L19 11z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-wk4wti-MuiTypography-root"
        >
          Teste Veículo
        </p>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-7qgm32-MuiTypography-root"
    >
      R$ 
      10
    </p>
  </div>
  <div
    class="MuiBox-root css-bo4obb"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-wk4wti-MuiTypography-root"
    >
      Teste Descrição
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p1tr6x-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Solicitar
    </button>
  </div>
</div>
`;
